src/Cli/dotnet/commands/dotnet-workload/install/WorkloadManifestUpdater.cs

```diff
@@ -0,0 +1,155 @@
+        private readonly INuGetPackageDownloader _nugetPackageDownloader;
+        private readonly string _userHome;
+
+        public WorkloadManifestUpdater(
+            IReporter reporter,
+            IWorkloadManifestProvider workloadManifestProvider,
+            INuGetPackageDownloader nugetPackageDownloader,
+            string userHome)
+        {
+            _reporter = reporter;
+            _workloadManifestProvider = workloadManifestProvider;
+            _userHome = userHome;
+            _nugetPackageDownloader = nugetPackageDownloader;
+        }
+
+        public async Task UpdateAdvertisingManifestsAsync(SdkFeatureBand featureBand)
+        {
+            var manifests = GetInstalledManifestIds();
+            foreach (var manifest in manifests)
+            {
+                await UpdateAdvertisingManifestAsync(manifest, featureBand);
+            }
+        }
+
+        public IEnumerable<(ManifestId manifestId, ManifestVersion existingVersion, ManifestVersion newVersion)> CalculateManifestUpdates(SdkFeatureBand featureBand)
+        {
+            var manifestUpdates = new List<(ManifestId, ManifestVersion, ManifestVersion)>();
+            var currentManifestIds = GetInstalledManifestIds();
+            foreach (var manifestId in currentManifestIds)
+            {
+                var currentManifestVersion = GetInstalledManifestVersion(manifestId);
+                var adManifestVersion = GetAdvertisingManifestVersion(featureBand, manifestId);
+                if (adManifestVersion == null)
+                {
+                    continue;
+                }
+
+                if (adManifestVersion != null && adManifestVersion.CompareTo(currentManifestVersion) > 0)
+                {
+                    manifestUpdates.Add((manifestId, currentManifestVersion, adManifestVersion));
+                }
+            }
+            return manifestUpdates;
+        }
+
+        private IEnumerable<ManifestId> GetInstalledManifestIds()
+        {
+            var manifestDirs = _workloadManifestProvider.GetManifestDirectories();
+
+            var manifests = new List<ManifestId>();
+            foreach (var manifestDir in manifestDirs)
+            {
+                var manifestId = Path.GetFileName(manifestDir);
+                manifests.Add(new ManifestId(manifestId));
+            }
+            return manifests;
+        }
+
+        private async Task UpdateAdvertisingManifestAsync(ManifestId manifestId, SdkFeatureBand featureBand)
+        {
+            string packagePath = null;
+            try
+            {
+                var adManifestPath = GetAdvertisingManifestPath(featureBand, manifestId);
+                packagePath = await _nugetPackageDownloader.DownloadPackageAsync(GetManifestPackageId(featureBand, manifestId));
+                var resultingFiles = await _nugetPackageDownloader.ExtractPackageAsync(packagePath, adManifestPath);
```

Comment:
How does this behave if there are already files there?  It should delete the existing files and replace them, I believe.

Reply:
Yes this should overwrite existing files.